endpoints/openrtb2/amp_auction_test.go

```diff
@@ -24,69 +25,71 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
-// From auction_test.go
-// const maxSize = 1024 * 256
-
 // TestGoodRequests makes sure that the auction runs properly-formatted stored bids correctly.
 func TestGoodAmpRequests(t *testing.T) {
```

Comment:
This is looking good. Nice work!
My only suggestions regarding this is to have every JSON file begin with the most simplistic valid base request that would go through the whole flow without using any stored responses. Then slightly tweak each test to verify the specific part that is under test. This would establish symmetry among the tests reducing confusion.

Some of the JSON files contain additional unrequired fields like `targeting`, `pmp.deals`, etc and it isn't clear why those fields are part of that test. If we actually want to make sure all of those unrequired fields don't break the AMP endpoint but they don't require their own dedicated JSON file tests (because it would result in an explosion in the number of JSON files), I suggest creating one JSON test with a fully loaded request, but perhaps even that isn't necessary.

As for establishing symmetry and improving intent, it seems like:
-  only `imp-with-stored-resp.json` should define `mockAmpStoredResponse`.
- `mock`gdpr.json` and `gdpr-no-consentstring.json` should be the same except for the consent string.
- `aliases.json` and `aliased-buyeruids.json` should be the same except for `user.ext.prebid.buyeruids`
- `user.json` and `aliased-buyeruids.json` should be the same except for `ext.prebid.aliases`

Reply:
> `aliases.json` and `aliased-buyeruids.json` should be the same except for `user.ext.prebid.buyeruids`
 
 Should we also swap the `"unknown"` to `"appnexus"` in the `"ext"` too? Or is that part of the test? Is the test trying to prove that PBS core understands that `"unknown"` found in the `"buyeruids"` field refers to `"appnexus"`?
```
19         "ext": {
20           "prebid": {
21               "storedauctionresponse": {
22                   "id": "6d718149"
23               }
24           },
25 -         "appnexus": {
   +         "unknown": {
26             "placementId": 12883451
27           }
28         }
29       }
30     ],
31     "user": {
32       "ext": {
33         "prebid": {
34           "buyeruids": {
35             "unknown": "123"
36           }
37         }
38       }
39     },
40     "ext": {
41       "prebid": {
42         "aliases": {
43           "unknown": "appnexus"
44         }
45       }
46     }
47   },
endpoints/openrtb2/sample-requests/valid-whole/supplementary/aliased-buyeruids.json
```

Reply:
> `user.json` and `aliased-buyeruids.json` should be the same except for `ext.prebid.aliases`
 
`user.json` didn't need the user part to pass. When removed, it looked exactly like `endpoints/openrtb2/sample-requests/valid-whole/supplementary/imp-with-stored-resp.json` which is the same as `aliased-buyeruids.json` except for `ext.prebid.aliases`. I removed `user.json`

 > mockgdpr.json and gdpr-no-consentstring.json should be the same except for the consent string.

Agree, reduced.

 > only `imp-with-stored-resp.json` should define `mockAmpStoredResponse`.

Agree, reduced.

`endpoints/openrtb2/sample-requests/valid-whole/supplementary/site-stored-response.json` without the clutter is the same as `endpoints/openrtb2/sample-requests/valid-whole/supplementary/imp-with-stored-resp.json`. The former got removed.

Reply:
> This is looking good. Nice work!
 
 Thanks Brian 😄